orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy

```diff
@@ -52,9 +52,13 @@ trait DeploymentDetailsAware {
   Stage getPreviousStageWithImage(Stage stage, String targetRegion, String targetCloudProvider) {
     getAncestors(stage).find {
       def regions = (it.context.region ? [it.context.region] : it.context.regions) as Set<String>
-      (it.context.containsKey("ami") || it.context.containsKey("amiDetails")) &&
-      (!targetRegion || regions?.contains(targetRegion)) &&
-      (targetCloudProvider == it.context.cloudProvider || targetCloudProvider == it.context.cloudProviderType)
+      def cloudProviderFromContext = it.context.cloudProvider ?: it.context.cloudProviderType
+      boolean hasTargetRegion = !targetRegion || regions?.contains(targetRegion)
+      boolean hasImage = it.context.containsKey("ami") || it.context.containsKey("amiDetails")
+      if (!cloudProviderFromContext) {
```

Comment:
Could go a step further by using:
`boolean hasTargetCloudProvider = !cloudProviderFromContext || targetCloudProvider == cloudProviderFromContext`
just after `def cloudProviderFromContext = ...`

And then all we'd need is:
`return hasImage && hasTargetRegion && hasTargetCloudProvider`